documentation/swagger-authoring-manual-guidelines.md

```diff
@@ -0,0 +1,52 @@
+| Terminology in Swagger and Azure Portal MUST be the same | Warning | |
+| Parameter names in the apis MUST be name of the resource like virtualMachines{virtualMachine} or virtualMachines{virtualMachineName} both are good example but virtualMachines{vm} is bad example | Warning | |
+| Mutability of properties MUST be marked up using [`x-ms-mutability` extension](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-mutability) | Warning | |
+| Error definitions MUST be abstracted, modeled and reused | Warning | |
+| A Swagger specification MUST use "x-ms-codegeneration-settings": { "name": "<ServiceName>Client" } for providing a better name to the generated client | Warning | SDK Violation |
+| "externalDocs" property MUST have both "url" and "description" accurately provided. More details can be found here | Warning | SDK Violation |
+| Odata parameters like "$filter" MUST document the properties on which the object can be filtered on and the supported operators in the description | Warning | SDK Violation |
+| Odata parameters like "$filter" MUST have an externalDocs that provides good examples of how the "$filter" parameter can be used | Warning | SDK Violation |
+| Parameters or properties that can only accept one value MUST be marked as "required" and they MUST have an "enum" array constraint with one valid value in it | Warning | SDK Violation |
+| If a parameter or a model property indicates that it is some "kind/type" of an artifact that can have a value from set of possible values, then it SHOULD have an "enum": ["Array of possible values"] constraint on that entity. It MUST also use the "x-ms-enum" extension. If creating an enum is not possible, then the values need to be documented in description. If the number of allowed values is huge, add a link to some kind of external documentation in the description, where those values are defined | Warning | SDK Violation |
+| If a swagger specification has a "fully" pageable operation and a "single-page" only operation then the generated code in C# will generate Page1.cs and Page.cs. They will differ on nextLink. To avoid this it is best advised that all the operations SHOULD defined as "fully" pageable in the specification | Warning | SDK Violation |
+| Swagger authoring MUST NOT be assigned to engineers who do not have an intimate knowledge of a service endpoint and its developer experience to avoid feeding inefficiencies into teams that focus on Azure developer experiences and the rest of the Azure eco-system | Warning | |
+| Descriptions MUST NOT contain spelling errors, grammatical errors, run off sentences and dummy texts | Warning | SDK Violation |
+| The description/summary MUST use correct English spelling and grammar | Warning | SDK Violation |
+| One MUST NOT use the API's code-symbol capitalization style for acronyms. For example, use "URL", NOT "Url"; and use "ID", NOT "Id" | Warning | SDK Violation |
```

Comment:
this may not be true, we're actually allowing ID, IP in the properties, https://msdn.microsoft.com/en-us/library/141e06ef(v=vs.71).aspx - URL is three letters, so it'd be Url, 2-letter acronyms are allowed.